fpmsyncd/routesync.cpp

```diff
@@ -1222,3 +1269,180 @@ string RouteSync::getNextHopWt(struct rtnl_route *route_obj)
 
     return result;
 }
+
+void RouteSync::sendOffloadReply(struct nlmsghdr* hdr)
+{
+    SWSS_LOG_ENTER();
+
+    assert(m_fpmInterface);
+
+    if (hdr->nlmsg_type != RTM_NEWROUTE)
+    {
+        return;
+    }
+
+    struct rtmsg *rtm = static_cast<struct rtmsg*>(NLMSG_DATA(hdr));
+
+    rtm->rtm_flags |= RTM_F_OFFLOAD;
+
+    // Send to zebra
+    if (!m_fpmInterface->send(hdr))
+    {
+        SWSS_LOG_ERROR("Failed to send reply to zebra");
+        return;
+    }
+}
+
+void RouteSync::sendOffloadReply(struct rtnl_route* route_obj)
+{
+    SWSS_LOG_ENTER();
+
+    nl_msg* msg{};
+    rtnl_route_build_add_request(route_obj, NLM_F_CREATE, &msg);
+
+    auto ownedMsg = std::unique_ptr<nl_msg, void(*)(nl_msg*)>(msg, nlmsg_free);
+
+    sendOffloadReply(nlmsg_hdr(ownedMsg.get()));
+}
+
+void RouteSync::setSuppressionEnabled(bool enabled)
+{
+    SWSS_LOG_ENTER();
+
+    m_isSuppressionEnabled = enabled;
+
+    SWSS_LOG_NOTICE("Pending routes suppression is %s", (m_isSuppressionEnabled ? "enabled": "disabled"));
+}
+
+void RouteSync::onRouteResponse(const std::string& key, const std::vector<FieldValueTuple>& fieldValues)
+{
+    IpPrefix prefix;
+    std::string vrfName;
+    std::string protocol;
+
+    bool isSetOperation{false};
+    bool isSuccessReply{false};
+
+    if (!isSuppressionEnabled())
+    {
+        return;
+    }
+
+    auto colon = key.find(':');
+    if (colon != std::string::npos && key.substr(0, colon).find("Vrf") != std::string::npos)
```

Comment:
"Vrf" -> define?

Reply:
There is no parsing of key from ROUTE_TABLE in fpmsyncd. It is present in orchagent where I have copied this logic from. But there is a macro for "Vrf"